src/Microsoft.AspNet.Razor/SourceLocation.cs

```diff
@@ -76,12 +125,13 @@ public static SourceLocation Advance(SourceLocation left, string text)
             return tracker.CurrentLocation;
         }
 
-        public static SourceLocation Add(SourceLocation left, SourceLocation right)
+        private static SourceLocation Add(SourceLocation left, SourceLocation right)
```

Comment:
implementation of this and `Subtract()` assumes the two locations are in the same file.  that case cries out for an `ArgumentException`.